tools/Azure.SDK.ChangelogGen/Azure.SDK.ChangelogGen/Context.cs

```diff
@@ -0,0 +1,160 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#nullable disable
+using System.Reflection;
+using System.Text.Json;
+using System.Text.RegularExpressions;
+using Azure.SDK.ChangelogGen.Compare;
+using Azure.SDK.ChangelogGen.Report;
+using Azure.SDK.ChangelogGen.Utilities;
+using Microsoft.Extensions.Configuration;
+
+namespace Azure.SDK.ChangelogGen
+{
+    public enum MergeMode
+    {
+        Group,
+        Line,
+        OverWrite,
+    };
+
+    public class Context
+    {
+        public string ReleaseVersion { get; private set; }
+        public string ReleaseDate { get; private set; }
+        public bool UpdateReleaseVersionDate { get; private set; }
+
+        public string ApiFile { get; private set; }
+        public string ApiFileGithubKey => GetGithubKey(ApiFile);
+
+        public string BaselineVersion { get; private set; }
+        public string BaselineVersionReleaseDate { get; private set; }
+        public string BaselineGithubTag { get; private set; }
+
+        public string ChangeLogMdFile => Path.Combine(PackageFolder, "CHANGELOG.md");
+        public bool OverwriteChangeLogMdFile { get; private set; }
+        public MergeMode MergeMode { get; private set; }
+        public bool LogSettings { get; private set; } = false;
+
+        public string RepoRoot { get; private set; }
+
+        private DirectoryInfo PackageDir { get; set; }
+        public string PackageFolder => PackageDir.FullName;
+        public string PackageName => PackageDir.Name;
+
+        public string AutorestMdFile => Path.Combine(PackageFolder, "src/autorest.md");
+        public string AutorestMdGithubKey => GetGithubKey(AutorestMdFile);
+
+        public string AzureCoreChangeLogGithubKey => "sdk/core/Azure.Core/CHANGELOG.md";
```

Comment:
We won't update Azure Core's change log. We need to access the Azure Core changelog is to figure out the Azure Core version referenced in baseline and current library so that we can know whether the Azure Core Dependency version has changed.